### PR TITLE
Redo the Wayland checks for Linux

### PR DESCRIPTION
**Pull request by orowith2os**
The current method of doing them is quite messy, and doesn't follow the suggestions inside of the Electron docs: https://www.electronjs.org/docs/latest/api/command-line#commandlineappendswitchswitch-value

I wrote this on my laptop while unable to test, but I hope it works. All sources I can find say it does.

Additionally there should an `if (check) && (othercheck)` but I'm not quite sure how TypeScript handles that so I nested these checks.

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -25,21 +25,22 @@ if (!app.requestSingleInstanceLock()) {
...
-        if (process.platform == "linux") {
-            if (process.env.XDG_SESSION_TYPE == "wayland") {
-                console.log("Wayland specific patches applied.");
-                app.commandLine.appendSwitch("ozone-platform=wayland");
-                if (process.env.XDG_CURRENT_DESKTOP == "GNOME") {
-                    app.commandLine.appendSwitch("enable-features=UseOzonePlatform,WaylandWindowDecorations");
-                } else {
-                    app.commandLine.appendSwitch("enable-features=UseOzonePlatform");
-                }
+    if (process.platform == "linux") {
```
Comment (greysilly7) [inline src/main.ts]: Why not just do 
`
if (process.platform === "linux" && process.env.XDG_SESSION_TYPE === "wayland") {...}`

Comment (orowith2os) [reply src/main.ts]: That was what I was looking for, I'm just not familiar with TypeScript so I didn't wanna try it. Thanks.

Comment (orowith2os) [reply src/main.ts]: Also, what's with the `===`? Wouldn't just `==` work?

Comment (Vendicated) [reply src/main.ts]: == is loose equality ("0" == 0), === is strict ("0" !== 0)

It doesn't matter here, but always using === (except when comparing to null [ == null is equivalent to === null || === undefined ]) is best practice
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness

Comment (orowith2os) [reply src/main.ts]: we might not want that then, assuming === makes Wayland and wayland seem different (if == doesn't already do that).

Comment (Vendicated) [reply src/main.ts]: === doesn't do that either, if you want case insensitive comparison you should call .toLowerCase() on both items

Comment (orowith2os) [reply src/main.ts]: It's not too much of an issue anyways, since AFAIK most (or all) set `wayland` and not `Wayland`.

Comment (greysilly7) [reply src/main.ts]: imo you should do the toLowerCase() so there is no "WhY NO WorKy"

Comment (orowith2os) [reply src/main.ts]: Alright, I think that's it.

Comment (orowith2os) [reply src/main.ts]: wait crap I think I did it wrong... could you write me a snippet or another review suggestion to do it?

File: src/main.ts [hunk 2]
```diff
@@ -25,21 +25,22 @@ if (!app.requestSingleInstanceLock()) {
...
-                console.log("Wayland specific patches applied.");
-                app.commandLine.appendSwitch("ozone-platform=wayland");
-                if (process.env.XDG_CURRENT_DESKTOP == "GNOME") {
-                    app.commandLine.appendSwitch("enable-features=UseOzonePlatform,WaylandWindowDecorations");
-                } else {
-                    app.commandLine.appendSwitch("enable-features=UseOzonePlatform");
-                }
-            }
+    // We use toLowerCase to account for desktops where XDG_SESSION_TYPE might be Wayland and not wayland.
+    if (process.platform === "linux".toLowerCase() && process.env.XDG_SESSION_TYPE === "wayland".toLowerCase()) {
```
Comment (Vendicated) [inline src/main.ts]:
```suggestion
    if (process.platform.toLowerCase() === "linux" && process.env.XDG_SESSION_TYPE.toLowerCase() === "wayland") {
```

**Comment by NextWork123** [comment]
Hello, tried to compile armcord with that pull request but nothing it gives me an error.

```
src/main.ts:29:55 - error TS18048: 'process.env.XDG_SESSION_TYPE' is possibly 'undefined'.

if (process.platform.toLowerCase() === "linux" && process.env.XDG_SESSION_TYPE.toLowerCase() === "wayland") {
                                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
Found 1 error in src/main.ts:29
```

**Comment by orowith2os** [comment]
Does it work without toLowerCase? That might be screwing with it.

**Comment by NextWork123** [comment]
Yep, is working but it have some trouble. Like that: 
Screenshot_13 i can't do anything it stay like that.

**Comment by orowith2os** [comment]
Are you on NVIDIA?

**Comment by NextWork123** [comment]
Nop, i have an AMD RX 6600.

**Comment by orowith2os** [comment]
Interesting. What if you run it with USE_WAYLAND=0? I'm also not sure if its an issue specific to your compositor or something else, like the distro.

**Comment by NextWork123** [comment]
Same result. I have that when debugging my distro: CachyOS

`[139634:0121/220226.144778:ERROR:shared_image_factory.cc(575)] Could not find SharedImageBackingFactory with params: usage: Gles2|Raster|DisplayRead|Scanout, format: BGRA_8888, share_between_threads: 0, gmb_type: shared_memory
[139634:0121/220226.146018:ERROR:shared_image_factory.cc(575)] Could not find SharedImageBackingFactory with params: usage: Gles2|Raster|DisplayRead|Scanout, format: BGRA_8888, share_between_threads: 0, gmb_type: shared_memory
[139634:0121/220226.147194:ERROR:shared_image_factory.cc(575)] Could not find SharedImageBackingFactory with params: usage: Gles2|Raster|DisplayRead|Scanout, format: BGRA_8888, share_between_threads: 0, gmb_type: shared_memory
[139634:0121/220226.148373:ERROR:shared_image_factory.cc(575)] Could not find SharedImageBackingFactory with params: usage: Gles2|Raster|DisplayRead|Scanout, format: BGRA_8888, share_between_threads: 0, gmb_type: shared_memory
[139634:0121/220226.149505:ERROR:shared_image_factory.cc(575)] Could not find SharedImageBackingFactory with params: usage: Gles2|Raster|DisplayRead|Scanout, format: BGRA_8888, share_between_threads: 0, gmb_type: shared_memory
[Config manager] trayIcon: default
[139634:0121/220226.412482:ERROR:shared_image_factory.cc(575)] Could not find SharedImageBackingFactory with params: usage: Gles2|Raster|DisplayRead|Scanout, format: BGRA_8888, share_between_threads: 0, gmb_type: shared_memory`

**Comment by orowith2os** [comment]
This could be an issue entirely unrelated to this PR, better to split it into a dedicated issue.

**Comment by Vendicated** [comment]
why did you remove the toLowerCase? you just need to make it null safe by using optional chaining like XDG_SESSION_TYPE?.toLowerCase()

**Comment by orowith2os** [comment]
@NextWork123 mind giving this one more shot? Probably won't fix the GUI itself bit builds should work now.

**Comment by NextWork123** [comment]
@orowith2os it compiled without any error. Only the GUI have some issue but i don't know

**Comment by orowith2os** [comment]
Thanks for testing. Make a separate issue for the GUI, because this PR doesn't change anything aside from how to check for Wayland. Any issues you have for the GUI should be reproducible on an upstream dev build.

**Comment by smartfrigde** [comment]
LGTM

**Comment by smartfrigde** [comment]
Reverting until #336 gets fixed

**Pull request merged**